MailNotificationServiceExtension/NotificationService.swift

```diff
@@ -30,13 +30,15 @@ final class NotificationService: UNNotificationServiceExtension {
     private let dependencyInjectionHook = EarlyDIHook()
 
     @LazyInjectService private var accountManager: AccountManager
+    @LazyInjectService private var mailboxInfosManager: MailboxInfosManager
 
     var contentHandler: ((UNNotificationContent) -> Void)?
     var bestAttemptContent: UNMutableNotificationContent?
 
     override init() {
         super.init()
         Logging.initLogging()
+        ModelMigrator().migrateRealmIfNeeded()
```

Comment:
I didn't check but please test that no call to realm is made before this call else it will be useless. Because EarlyDIHook is called before init and maybe accountManager can be called before ?

Reply:
I'll check, the idea was exactly that, to do it early before we handle notifications (in realm) … but with DI in place.